includes/class-wc-payments-checkout.php

```diff
@@ -90,16 +93,51 @@ public function __construct(
 	 * @return void
 	 */
 	public function init_hooks() {
-		add_action( 'wc_payments_add_payment_fields', [ $this, 'payment_fields' ] );
+		add_action( 'wc_payments_set_gateway', [ $this, 'set_gateway' ] );
+		add_action( 'wc_payments_add_upe_payment_fields', [ $this, 'payment_fields' ] );
+		add_action( 'woocommerce_after_account_payment_methods', [ $this->gateway, 'remove_upe_setup_intent_from_session' ], 10, 0 );
+		add_action( 'woocommerce_subscription_payment_method_updated', [ $this->gateway, 'remove_upe_setup_intent_from_session' ], 10, 0 );
+		add_action( 'woocommerce_order_payment_status_changed', [ get_class( $this->gateway ), 'remove_upe_payment_intent_from_session' ], 10, 0 );
+		add_action( 'wp', [ $this->gateway, 'maybe_process_upe_redirect' ] );
+		add_action( 'wc_ajax_wcpay_log_payment_error', [ $this->gateway, 'log_payment_error_ajax' ] );
+		add_action( 'wp_ajax_save_upe_appearance', [ $this->gateway, 'save_upe_appearance_ajax' ] );
+		add_action( 'wp_ajax_nopriv_save_upe_appearance', [ $this->gateway, 'save_upe_appearance_ajax' ] );
+		add_action( 'switch_theme', [ $this->gateway, 'clear_upe_appearance_transient' ] );
+		add_action( 'woocommerce_woocommerce_payments_updated', [ $this->gateway, 'clear_upe_appearance_transient' ] );
+		add_action( 'wc_ajax_wcpay_init_setup_intent', [ $this->gateway, 'init_setup_intent_ajax' ] );
+		add_action( 'wc_ajax_wcpay_log_payment_error', [ $this->gateway, 'log_payment_error_ajax' ] );
+
+		add_action( 'wp_enqueue_scripts', [ $this, 'register_scripts' ] );
 		add_action( 'wp_enqueue_scripts', [ $this, 'register_scripts_for_zero_order_total' ], 11 );
 	}
 
 	/**
-	 * Enqueues and localizes WCPay's checkout scripts.
+	 * Registers all scripts, necessary for the gateway.
 	 */
-	public function enqueue_payment_scripts() {
-		wp_localize_script( 'WCPAY_CHECKOUT', 'wcpayConfig', WC_Payments::get_wc_payments_checkout()->get_payment_fields_js_config() );
-		wp_enqueue_script( 'WCPAY_CHECKOUT' );
```

Comment:
Thanks for asking. This code was only used in the [legacy card gateway/checkout](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-checkout.php#L93) in the past. With UPE & also deferred UPE, [other implementation is loaded](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-upe-checkout.php#L94) which adds the `wcpay_upe_config` JS object [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-upe-checkout.php#L298-L303) and [front-end searches for this object if it can't find `wcpayConfig`.](https://github.com/Automattic/woocommerce-payments/blob/6.6.0/client/utils/checkout.js#L14-L16). 

Once we finish the most critical cleanup parts, I'll look into consolidating these constants & creating issues.

Reply:
That makes sense. Thanks for the explanation!